### PR TITLE
Use hostname in the service model abstraction

### PR DESCRIPTION
**Pull request by kyessenov**
PR removes "Namespace" from Istio service, and instead uses fully qualified "Hostname" property.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/manager/66)

### CODE REVIEW

File: platform/kube/controller.go [hunk 1]
```diff
@@ -420,15 +442,27 @@ func convertService(svc v1.Service) *model.Service {
...
+		Address:  addr,
 		// TODO: empty set of service tags for now
 		Tags: nil,
 	}
 }
 
+// parseHostname is the inverse of hostname pattern from Kubernetes/Istio service translation
+func parseHostname(hostname string) (string, string, error) {
+	prefix := strings.TrimSuffix(hostname, "."+ServiceSuffix)
+	if len(prefix) < len(hostname) {
```
Comment (ayj) [inline platform/kube/controller.go]: nit: Consider reducing the indentation level here by returning early on errors. See https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

**Comment by ayj** [comment]
:lgtm:

---

Review status: 0 of 10 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/istio/manager/66#-:-Kbm2ofYCd8Q8I8o_ck4:bnfp4nl)*

File: platform/kube/controller.go [hunk 2]
```diff
@@ -51,6 +51,11 @@ type cacheHandler struct {
 	handler  *chainHandler
 }
 
+const (
+	// ServiceSuffix is the hostname suffix used by a Kubernetes service
+	ServiceSuffix = "svc.cluster.local"
```
Comment (rshriram) [inline platform/kube/controller.go]: Nit: May be add a todo here to make this configurable in future? I vaguely remember having the ability to define the suffix when initializing the kubernetes cluster. If that has changed , then please disregard this comment.

File: model/service.go [hunk 3]
```diff
@@ -26,8 +26,8 @@ type ServiceDiscovery interface {
 	// Services list declarations of all services and their tags
 	Services() []*Service
 
-	// GetService retrieves a service by name if it exists
-	GetService(name, namespace string) (*Service, bool)
+	// GetService retrieves a service by host name if it exists
+	GetService(hostname string) (*Service, bool)
```
Comment (rshriram) [inline model/service.go]: This is clean. Makes the model pretty agnostic to underlying platform.

File: platform/kube/controller.go (hunk 2 above)
Comment (kyessenov) [inline platform/kube/controller.go]: Yeah, I did not know the DNS suffix is configurable.

File: platform/kube/controller.go (hunk 1 above)
Comment (kyessenov) [inline platform/kube/controller.go]: ok, PTAL.

**Comment by ayj** [comment]
:lgtm:

---

Review status: 0 of 10 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/istio/manager/66#-:-KbmPiBFpULpR4_nSRAl:bnfp4nl)*

**Comment by ayj** [comment]
Reviewed 9 of 10 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/istio/manager/66)*

**Comment by kyessenov** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[model/service.go, line 30 at r1](https://reviewable.io:443/reviews/istio/manager/66#-KblLvW5Rtr4z9JfYF8p-r1-30:-KbmPuaL5nyoJkf6Qn2i:b-k73z76) ([raw file](https://github.com/istio/manager/blob/43e90b94f4e48b8fd23f0d7a8de59664e53b3663/model/service.go#L30)):*
<details><summary><i>Previously, rshriram (Shriram Rajagopalan) wrote…</i></summary><blockquote>

This is clean. Makes the model pretty agnostic to underlying platform. 
</blockquote></details>

thanks

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/istio/manager/66)*

**Comment by kyessenov** [comment]
Review status: 9 of 10 files reviewed at latest revision, 2 unresolved discussions.

---

*[model/service.go, line 30 at r1](https://reviewable.io:443/reviews/istio/manager/66#-KblLvW5Rtr4z9JfYF8p-r1-30:-KbmfUpC4GljFhmX4-nn:b-896fix) ([raw file](https://github.com/istio/manager/blob/43e90b94f4e48b8fd23f0d7a8de59664e53b3663/model/service.go#L30)):*
<details><summary><i>Previously, kyessenov (Kuat) wrote…</i></summary><blockquote>

thanks
</blockquote></details>

Done.

---

*[platform/kube/controller.go, line 56 at r1](https://reviewable.io:443/reviews/istio/manager/66#-KblLvW5Rtr4z9JfYF8s-r1-56:-KbmfRT-6RALIEpRnx17:b-896fix) ([raw file](https://github.com/istio/manager/blob/43e90b94f4e48b8fd23f0d7a8de59664e53b3663/platform/kube/controller.go#L56)):*
<details><summary><i>Previously, kyessenov (Kuat) wrote…</i></summary><blockquote>

Yeah, I did not know the DNS suffix is configurable.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/istio/manager/66)*

**Pull request merged**